python/lsst/ts/salobj/base_csc.py

```diff
@@ -148,7 +180,37 @@ async def start(self):
         self._heartbeat_task = asyncio.create_task(self._heartbeat_loop())
         await self.set_simulation_mode(self._requested_simulation_mode)
 
+        # Handle initial state, then transition to the desired state.
+        # If this fails then log the exception and continue,
+        # in hopes that the CSC can still be used. For instance
+        # if the settings were invalid, the user can specify others.
         await self.handle_summary_state()
+        command = None
+        if self._initial_state != self.default_initial_state:
+            state_transition_dict = make_state_transition_dict()
+            # List of commands to transition from
+            # current state to initial_state.
+            # Use the current state instead of default_initial_state
+            # because of Hexapod and Rotator, which do not know their
+            # current state until they connect to the low-level controller.
+            command_state_list = state_transition_dict[
+                (self.summary_state, self._initial_state)
+            ]
+            state_transition_commands = [item[0] for item in command_state_list]
+            try:
+                self._enable_summary_state_event = False
```

Comment:
I don't like the idea of suppressing the summary state events for automatic transitions. If I look at EFD records I should find no difference between an automatic state changer and a commanded CSC. Otherwise, it looks like the automatic one magically got to that state.

Reply:
I agree it makes me a bit nervous as well. But I think we need it, or a new generic event, instead.

The idea is that when the user sees a summaryState even then they know that the CSC will respond to SAL commands. Without such a rule a user has no idea when the CSC has finished starting up and will listen to SAL commands.

This has always been an issue, but it is made much worse by the fact that the CSC may go through several states before it is ready to be commanded.

It also reduces unit test breakage because we don't get extra summary state messages. (I could probably modify BaseCscTestCase to ameliorate the problem).

Reply:
As agreed at a "standup" meeting, I modified the code as follows:
* Do NOT suppress summaryState events (addressing your concern).
* Enable command callbacks as soon as the first summaryState is output (before that the CSC is likely to be only partially running and commands may fail for confusing reasons).
* Update BaseCscTestCase.make_csc to check all except the final summary state. This will eliminate code breakage in tests that use that method to create CSCs (which should be most of them).